api/dao/liststorage.py

```diff
@@ -0,0 +1,173 @@
+import re
+import copy
+import logging
+import datetime
+import bson.errors
+import bson.objectid
+
+from .. import mongo
+from . import APIStorageException
+
+log = logging.getLogger('scitran.api')
+
+
+class ListStorage(object):
+    """
+    This class provides access to sublists of a mongodb collections elements (called containers).
+    """
+
+    def __init__(self, cont_name, list_name, use_oid = False):
+        self.cont_name = cont_name
+        self.list_name = list_name
+        self.use_oid = use_oid
+        self.dbc = mongo.db[cont_name]
+
+    def get_container(self, _id, query_params=None):
+        """
+        Load a container from the _id.
+
+        This method is usually used to to check permission properties of the container.
+        e.g. list of users that can access the container
+
+        For simplicity we load its full content.
+        """
+        if self.use_oid:
+            _id = bson.objectid.ObjectId(_id)
+        query = {'_id': _id}
+        projection = None
+        if query_params:
+            query[self.list_name] = {
+                '$elemMatch': query_params
```

Comment:
Good question:
actually the variable name is kind of misleading and needs to be changed.
the query_params are not the request query_params
but essentially they are the variable extracted in the route using the regexes.
As they are mapped from the **kwargs in the handler methods.
https://webapp-improved.appspot.com/guide/routing.html#the-url-template

So that is were the validation happens: in the routing table.